core/client/fs/src/main/java/alluxio/client/ByteBufferReadable.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+package alluxio.client;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+/**
+ * Implementers of this interface provide a read API that writes to a ByteBuffer, not a byte[].
+ */
+public interface ByteBufferReadable {
```

Comment:
Check `alluxio.client.file.FileInStream` where we already define and provided the default implementation of `  public int read(ByteBuffer byteBuffer, int off, int len) throws IOException`.

You can either move this method there (and remove interface `ByteBufferReadable`), or move that methd here. I don't have a strong opinion. But I do think they should be put together

Reply:
Yeah, as there already one `public int read(ByteBuffer byteBuffer, int off, int len) throws IOException` method in `alluxio.client.file.FileInStream`, so I think it's ok to move the interface's logic to `alluxio.client.file.FileInStream` class.